cmd/clientapi/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"flag"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary/client_api/api"
+	"github.com/docker/notary/cmd/clientapi/setup"
+	"github.com/docker/notary/utils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	configPath string
+)
+
+func init() {
+	flag.StringVar(
+		&configPath,
+		"config",
+		"config.toml",
+		"path to configuration file; supported formats are JSON, YAML, and TOML",
+	)
+	flag.Parse()
+}
+
+func main() {
+	vc, err := setup.ViperConfig(configPath)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	logrus.SetLevel(logrus.DebugLevel)
+
+	// FIXME: refactor with notary-signer/config.go:getAddrAndTLSConfig()
+	// and setupGRPCServer()
+	grpcAddr, tlsConfig, err := setup.GetAddrAndTLSConfig(vc)
+	if err != nil {
+		logrus.Fatal("unable to set up TLS: %s", err.Error())
```

Comment:
Worth noting HTTP2 _requires_ TLS. It'll generate self signed certs or some such if none are provided. It's entirely up to the client to set itself to insecure mode.